File: LR_5-6/блок 3/pack_3_zadacha_2.py
import random
arr = []
n = 20
print('Исходный массив:')
for i in range(n):
    arr.append(random.randint(-50, 50))
    print(arr[i], end=' ')
print()
j = 0
for i in range(n):
    if arr[i] >= 0:
        arr[j] = arr[i]
        j += 1
print('Массив после удаления:')
if j > 0:
    for i in range(j):
        print(arr[i], end=' ')
    print()
    print(f'Количество элементов: {j}')
else:
    print('Все элементы были отрицательными')

File: LR_5-6/блок 3/pack_3_zadacha_2.pas
program zadacha_2;
uses crt;
var arr: array [1..50] of integer;
    i, j,n:integer;
begin
  randomize;
    n:=20;
    writeln ('исходный массив: ');
    for i:=1 to n do
    begin
      arr[i]:=random(101)-50;
      write(arr[i], ' ');
    end;
    writeln;
    j:=0;
    for i:=1 to n do
    begin
      if arr[i]>=0 then
      begin
        j:=j+1;
        arr[j]:=arr[i];
      end;
    end;
    write('массив после удаления: ');
    if j>0 then
    begin
      for i:=1 to j do
        write(arr[i], ' ');
      writeln;
      writeln('количество элементов: ',j);
    end
    else writeln('все элементы были отрицательными');
end.